src/instructlab/sdg/generate_data.py

```diff
@@ -418,3 +420,9 @@ def generate_data(
 
     generate_duration = time.time() - generate_start
     logger.info(f"Generation took {generate_duration:.2f}s")
+    if len(empty_sdg_leaf_nodes) > 0:
+        logger.warning(
+            "Leaf nodes with empty sdg output: {}".format(
+                " ".join(empty_sdg_leaf_nodes)
```

Comment:
From testing this manually locally:
```
  File "/home/bbrownin/src/instructlab/sdg/venv/lib/python3.11/site-packages/instructlab/data/generate.py", line 293, in generate
    generate_data(
  File "/home/bbrownin/src/instructlab/sdg/src/instructlab/sdg/generate_data.py", line 426, in generate_data
    " ".join(empty_sdg_leaf_nodes)
TypeError: sequence item 0: expected str instance, Dataset found
```

Reply:
I pushed a new commit to address this issue, as well as add a test to confirm that we don't error and log a proper warning message at the end of the generation. So, let's reverse our roles and you review my change - feel free to amend, rebase, etc as desired. Thanks!